### PR TITLE
RHPAM-3897: switch to jboss-transaction-spi-jakarta

### PR DESCRIPTION
**Pull request by almope**
Signed-off-by: Alberto Morales Perez <almorale@redhat.com>

**JIRA**: [RHPAM-3897](https://issues.redhat.com/browse/RHPAM-3897)

**referenced Pull Requests**:
* https://github.com/kiegroup/jbpm/pull/2016
* https://github.com/kiegroup/drools/pull/3832
* https://github.com/kiegroup/droolsjbpm-integration/pull/2602
* https://github.com/kiegroup/jbpm-work-items/pull/215
* https://github.com/kiegroup/kie-jpmml-integration/pull/64
* https://github.com/kiegroup/optaplanner/pull/1574
* https://github.com/kiegroup/kie-soup/pull/227
* https://github.com/jboss-integration/rhba-installers/pull/92

### CODE REVIEW

**Comment by almope** [comment]
Jenkins run fdb

File: narayana-integration-bom/pom.xml [hunk 1]
```diff
@@ -40,18 +40,18 @@
         <groupId>org.jboss.jbossts.jta</groupId>
         <artifactId>narayana-jta</artifactId>
         <version>${version.org.jboss.jbossts.jta}</version>
-        <exclusions>
```
Comment (almope) [inline narayana-integration-bom/pom.xml:43]: This was here by mistake I think. As there's no dependency on this on this `narayana-jta`.

Comment (mareknovotny) [reply narayana-integration-bom/pom.xml:43]: maybe it was there as transitive dependency in past

File: narayana-integration-bom/pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@
     <!-- DBCP connection pooling for Narayana -->
     <version.org.apache.tomcat.tomcat-dbcp>9.0.21</version.org.apache.tomcat.tomcat-dbcp>
 
-    <version.org.jboss.transaction.spi>7.6.0.Final</version.org.jboss.transaction.spi>
+    <version.org.jboss.transaction.spi>7.6.1.Final</version.org.jboss.transaction.spi>
     <version.jakarta.transaction-api>1.3.3</version.jakarta.transaction-api>
```
Comment (gmunozfe) [inline narayana-integration-bom/pom.xml:30]: _jboss-transaction-spi-jakarta_ pom has different versions for transaction-api and resource-api that are being downgraded here:

https://github.com/jbosstm/jboss-transaction-spi/blob/7.6.1.Final/jakarta/pom.xml#L30-L31

**Comment by almope** [comment]
Jenkins run fdb

**Comment by almope** [comment]
jenkins run fdb

File: narayana-integration-bom/pom.xml (hunk 2 above)
Comment (almope) [reply narayana-integration-bom/pom.xml:30]: FDB failed for unrelated errors. The downgraded versions for transaction-api and resource-api seem to integrate well with jboss-transaction-spi 7.6.1.Final version.

**Comment by mareknovotny** [comment]
jenkins do fdb

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**